lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/commits.js

```diff
@@ -0,0 +1,115 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var join = require( 'path' ).join;
+var spawn = require( 'child_process' ).spawnSync; // eslint-disable-line node/no-sync
+var logger = require( 'debug' );
+var replace = require( '@stdlib/string/replace' );
+var trim = require( '@stdlib/string/trim' );
+
+
+// VARIABLES //
+
+var debug = logger( 'changelog:parse-commits' );
+var GIT_COMMIT_SEP = '^---^';
+
+
+// FUNCTIONS //
+
+/**
+* Escapes double quotes in a string.
+*
+* @private
+* @param {string} str - input string
+* @returns {string} escaped string
+*
+* @example
+* var str = escapeDoubleQuotes( 'beep "boop"' );
+* // returns 'beep \"boop\"'
+*/
+function escapeDoubleQuotes( str ) {
+	return replace( str, '"', '\\"' );
+}
+
+
+// MAIN //
+
+/**
+* Extracts commit details for a specified directory.
+*
+* @private
+* @param {string} dir - directory for which to extract commit details
+* @returns {Array} array of commit details
+*
+* @example
+* var commits = extractCommits( '.' );
+* // returns [...]
+*/
+function extractCommits( dir ) {
+	var commits;
+	var lines;
+	var parts;
+	var args;
+	var opts;
+	var cmd;
+	var out;
+	var i;
+
+	opts = {
+		'maxBuffer': 1024*1024*60, // 60 MB
+		'stdio': ['pipe', 'pipe', 'ignore'], // stdin, stdout, stderr
+		'env': {
+			'GIT_COMMIT_SEP': GIT_COMMIT_SEP
+		}
+	};
+	cmd = join( __dirname, '..', 'scripts', 'commits.sh' );
+	commits = [];
+	try {
+		args = [ resolve( dir ) ];
+		out = spawn( cmd, args, opts );
+		out = trim( out.stdout.toString() );
+	} catch ( err ) {
+		debug( 'Encountered an error when attempting to extract commits: %s', err.message );
+		return commits;
+	}
```

Comment:
```suggestion
	args = [ resolve( dir ) ];
	try {
		out = spawn( cmd, args, opts );
	} catch ( err ) {
		debug( 'Encountered an error when attempting to extract commits: %s', err.message );
		return commits;
	}
	out = trim( out.stdout.toString() );
```

Only wrap logic in a try-catch which can reasonably be expected to potentially error. Otherwise, you can mask syntax errors, etc.